src/transports/httpclient.c

```diff
@@ -598,6 +598,7 @@ static int apply_credentials(
 	} else if (!token.size) {
 		git_error_set(GIT_ERROR_HTTP, "failed to respond to authentication challenge");
 		error = GIT_EAUTH;
+		error = -1;
```

Comment:
This doesn't look right. I think we want `error = GIT_EAUTH`, but we definitely don't need both assignments. :)

Reply:
Yep, agreed.  Fixed in main.